xml/System.Runtime.Loader/AssemblyLoadContext.xml

```diff
@@ -20,8 +20,45 @@
+ 
+ The <xref:System.Runtime.Loader.AssemblyLoadContext> exist primarily exists to provide assembly isolation. It allows multiple versions of the same assembly to be loaded within a single process. It replace the isolation mechanisms provided by multiple <xref:System.AppDomain> instances in the .NET Framework.
+
+### Usage in the runtime
+ The runtime implements two <xref:System.Runtime.Loader.AssemblyLoadContext>:
+     * The <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> represents the runtimes default binding context which is used for the framework and the application main assembly and its normal dependencies.
+     * The <xref:System.Reflection.Assembly.LoadFile(System.String)?displayProperty=nameWithType> method isolates the assemblies it loads by instantiating the most basic <xref:System.Runtime.Loader.AssemblyLoadContext>.  It has a simplistic isolation scheme which loads each assembly in its own <xref:System.Runtime.Loader.AssemblyLoadContext> with no dependency resolution.
+ 
+ To implement the <xref:System.Reflection.Assembly.LoadFrom(System.String)?displayProperty=nameWithType> event, the runtime adds a <xref:System.Runtime.Loader.AssemblyLoadContext.Resolving> event handler to bind to dependent files in the same directory.
+
+### Application usage
+ An application can create its own <xref:System.Runtime.Loader.AssemblyLoadContext> to create a custom solution for advanced binding scenarios. The customization focuses on defining dependency resolution mechanisms.
+ 
+ The <xref:System.Runtime.Loader.AssemblyLoadContext> provides two extension points to implement managed assembly resolution.
+     1. The <xref:System.Runtime.Loader.AssemblyLoadContext.Load(System.Reflection.AssemblyName)?displayProperty=nameWithType> provides the first chance for the <xref:System.Runtime.Loader.AssemblyLoadContext> to resolve the assembly, load it and return it. If the <xref:System.Runtime.Loader.AssemblyLoadContext.Load(System.Reflection.AssemblyName)?displayProperty=nameWithType> returns `null`, the loader will try to load the assembly into the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType>.
+     2. If the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> is unable to resolve the assembly, the original <xref:System.Runtime.Loader.AssemblyLoadContext> gets a second chance to resolve the assembly. The runtime raises the <xref:System.Runtime.Loader.AssemblyLoadContext.Resolving> event.
+ 
+ Additionally the <xref:System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDll(System.String)?displayProperty=nameWithType> virtual method allows customization of the default unmanaged assembly resolution. The default implementation is often sufficient for most typical scenarios.
+
+### Testing challenges
```

Comment:
I would call this "Implementation challanges" - it's not just testing, but also the implementation itself has to be somewhat careful and has to be prepared to be called more or less at any time.

Reply:
I am choosing "Technical Challenges" and combining the section with "Limitations"